Source/Relativity.Test.Helpers/Relativity.Test.Helpers/Kepler/OAuth2Helper.cs

```diff
@@ -0,0 +1,164 @@
+﻿using kCura.Relativity.Client;
+using Newtonsoft.Json.Linq;
+using Relativity.Services.Security;
+using Relativity.Services.Security.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using DTOs = kCura.Relativity.Client.DTOs;
+
+namespace Relativity.Test.Helpers.Kepler
+{
+	public class OAuth2Helper : IOAuth2Helper
+	{
+		private readonly IOAuth2ClientManager _oAuth2ClientManager;
+		private readonly IRSAPIClient _rsapiClient;
+
+		public OAuth2Helper(IOAuth2ClientManager oAuth2ClientManager, IRSAPIClient rsapiClient)
+		{
+			_oAuth2ClientManager = oAuth2ClientManager;
+			_rsapiClient = rsapiClient;
+		}
+
+		/// <summary>
+		/// Use this to create an OAuth2 authentication for Relativity, which will allow you to get a Bearer Token and gain access to Relativity
+		/// </summary>
+		/// <param name="username"></param>
+		/// <param name="oAuth2Name"></param>
+		/// <returns></returns>
+		public async Task<Services.Security.Models.OAuth2Client> CreateOAuth2ClientAsync(string username, string oAuth2Name)
+		{
+			DTOs.User user = GetUser(username);
+
+			List<Services.Security.Models.OAuth2Client> oAuth = await _oAuth2ClientManager.ReadAllAsync();
+			if (oAuth.Exists(x => x.Name.Equals(oAuth2Name)))

```

Comment:
this should probably be oAuth.Any(). does that makes sense?

Reply:
I ReadAllAsync and find a match on the user and auth name in question.  Multiple oAuth credentials can exist on a server at once so using Any would cause some issues